components/Messages.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "react";
+import Chat from "twilio-chat";
+
+export class Messages extends React.Component<
+    {
+        channel: any;
+        user: { id: string; username: string };
+    },
+    {
+        messages: any[];
+        newMessage: string;
+    }
+> {
+    constructor() {
+        // @ts-ignore
+        super(...arguments);
+        this.state = {
+            messages: [],
+            newMessage: "",
+        };
+    }
+
+    public async componentWillMount() {
+        const messagePage = await this.props.channel.getMessages();
+        this.setState({ messages: messagePage.items });
+        await this.props.channel.on("messageAdded", this.messageAdded);
+    }
+
+    public componentWillUnmount() {
+        this.props.channel.removeListener("messageAdded", this.messageAdded);
+    }
+
+    public async componentWillReceiveProps(nextProps: any) {
+        // If we change channel nextProps !== this.props so we have to load messages for the new channel
+        if (nextProps !== this.props) {
```

Comment:
I don't think it works as you are comparing 2 objects, even if looks the same, they are always different, for example:

```js
console.log({foo:true}==={foo:true}) // false
```

Reply:
Ok, i'll change it